test/models/user_test.rb

```diff
@@ -3,7 +3,38 @@
 require 'test_helper'
 
 class UserTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  test '#name_or_email' do
+    user = User.new(email: 'foo@example.com', name: '')
+    assert_equal 'foo@example.com', user.name_or_email
+
+    user.name = 'Foo.Bar'
+    assert_equal 'Foo.Bar', user.name_or_email
+  end
+
+  test '#follow_to_#following?' do
+    alice = users(:alice)
+    bob = users(:bob)
+
+    assert_not alice.following?(bob)
+    alice.follow(bob)
+    assert alice.following?(bob)
+  end
+
+  test '#followed_by?' do
+    alice = users(:alice)
+    bob = users(:bob)
+
+    assert_not alice.followed_by?(bob)
+    bob.active_relationships.find_or_create_by!(following_id: alice.id)
+    assert alice.followed_by?(bob)
+  end
+
+  test '#unfollow' do
+    alice = users(:alice)
+    bob = users(:bob)
+
+    alice.active_relationships.find_or_create_by!(following_id: bob.id)
+    alice.unfollow(bob)
+    assert_not alice.active_relationships.where(following_id: bob.id).exists?
```

Comment:
follow等のように、unfollowも以下のように修正するべきかと思っています。
ただし、2つのみならず3つの内容をテストすることになるため、これでいいのか悩んでいます・・・。
具体的には、以下のように修正しようと思っているのですが、こちら問題ないでしょうか？
```suggestion
    bob.follow(alice)
    assert alice.following?(bob)
    alice.unfollow(bob)
    assert_not alice.following?(bob)
```

Reply:
本来はあまりよくないですが、`alice.active_relationships.where(following_id: bob.id).exists?`のような長いコードを書くよりはベターだと思うので、まとめてしまっていいと思います。

ここに限っては`test "follow/unfollow/following?/followed_by?" do`みたいな感じで4つのメソッドをまとめてテストするような形でもいいんじゃないでしょうか。

Reply:
> 本来はあまりよくないですが、alice.active_relationships.where(following_id: bob.id).exists?のような長いコードを書くよりはベターだと思うので、まとめてしまっていいと思います。

確かにそうですね・・・。こちらまとめようと思います！